lib/runner.js

```diff
@@ -96,7 +96,9 @@ Runner.prototype.run = function () {
 		testCount: 0
 	};
 
-	return Promise.resolve(this.tests.buildPhases(this._addTestResult, this._bail).run()).then(function () {
+	this.tests.on('test', this._addTestResult);
+
+	return Promise.resolve(this.tests.build(this._bail).run()).then(function () {
```

Comment:
> I doubt that'll be noticeable given the cost of spawning a new process in order to run the tests.

It's [definitely noticeable](https://github.com/sindresorhus/ava/pull/466#issuecomment-174238246) in real test suites.

IMO, we need to stop comparing to "the cost of spawning a new process" as a reason to disqualify optimizations. That is a huge cost, nearly every optimization is going to be small in comparison at this point. There is a large subset of test suites for which AVA is demonstrably slower than the alternatives, and we are not going to close that gap without placing a high priority on optimizations.